framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1850,6 +1980,152 @@ NonlinearSystemBase::constraintJacobians(bool displaced)
       }
     }
   }
+
+  // go over NodeELemConstraints
+  std::set<dof_id_type> unique_slave_node_ids;
+  constraints_applied = false;
+  for (const auto & slave_id : _mesh.meshSubdomains())
+  {
+    for (const auto & master_id : _mesh.meshSubdomains())
+    {
+      if (_constraints.hasActiveNodeElemConstraints(slave_id, master_id, displaced))
+      {
+        const auto & constraints =
+            _constraints.getActiveNodeElemConstraints(slave_id, master_id, displaced);
+
+        // get unique set of ids of all nodes on current block
+        unique_slave_node_ids.clear();
+        const MeshBase & meshhelper = _mesh.getMesh();
+        for (const auto & elem : as_range(meshhelper.active_subdomain_elements_begin(slave_id),
+                                          meshhelper.active_subdomain_elements_end(slave_id)))
+        {
+          for (auto & n : elem->node_ref_range())
+            unique_slave_node_ids.insert(n.id());
+        }
+
+        for (auto slave_node_id : unique_slave_node_ids)
+        {
+          const Node & slave_node = _mesh.nodeRef(slave_node_id);
+          // check if slave node is on current processor
+          if (slave_node.processor_id() == processor_id())
+          {
+            // This reinits the variables that exist on the slave node
+            _fe_problem.reinitNodeFace(&slave_node, slave_id, 0);
+
+            // This will set aside residual and jacobian space for the variables that have dofs
+            // on the slave node
+            _fe_problem.prepareAssembly(0);
+            _fe_problem.reinitOffDiagScalars(0);
+
+            for (const auto & nec : constraints)
+            {
+              if (nec->shouldApply())
+              {
+                nec->_jacobian = &jacobian;
+
+                nec->computeConstraintForce();
```

Comment:
I don't think we actually even need to call this here because the force is not used in the Jacobian calculation.